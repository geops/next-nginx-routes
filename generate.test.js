"use strict";

const { execSync } = require("child_process");
const { readFileSync } = require("fs");

test("generate routes for example project", () => {
  execSync("cd example && npm run build");
  expect(readFileSync("./example/next-routes.conf", "utf8")).toMatchSnapshot();
});

test("generate routes for example project with custom basePath", () => {
  execSync("cd example && export BASE_PATH=/baz && npm run build");
  expect(readFileSync("./example/next-routes.conf", "utf8")).toMatchSnapshot();
});

test("generate routes for example project with trailingSlash enabled", () => {
  execSync("cd example && export TRAILING_SLASH=true && npm run build");
  expect(readFileSync("./example/next-routes.conf", "utf8")).toMatchSnapshot();
});
