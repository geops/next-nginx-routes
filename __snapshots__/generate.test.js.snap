// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate routes for example project 1`] = `
"
location ~ ^/(?:/)?$ {
    try_files /index.html /index.html;
}

location ~ ^/hello(?:/)?$ {
    try_files /hello.html /index.html;
}

location ~ ^/hello/world(?:/)?$ {
    try_files /hello/world.html /index.html;
}

location ~ ^/([^/]+?)(?:/)?$ {
    try_files /[foo].html /index.html;
}

location ~ ^/([^/]+?)/([^/]+?)/hello(?:/)?$ {
    try_files /[foo]/[bar]/hello.html /index.html;
}

location ~ ^/([^/]+?)/([^/]+?)/([^/]+?)(?:/)?$ {
    try_files /[foo]/[bar]/[id].html /index.html;
}"
`;

exports[`generate routes for example project with custom basePath 1`] = `
"
location ~ ^/baz(?:/)?$ {
    try_files /index.html /index.html;
}

location ~ ^/baz/hello(?:/)?$ {
    try_files /hello.html /index.html;
}

location ~ ^/baz/hello/world(?:/)?$ {
    try_files /hello/world.html /index.html;
}

location ~ ^/baz/([^/]+?)(?:/)?$ {
    try_files /[foo].html /index.html;
}

location ~ ^/baz/([^/]+?)/([^/]+?)/hello(?:/)?$ {
    try_files /[foo]/[bar]/hello.html /index.html;
}

location ~ ^/baz/([^/]+?)/([^/]+?)/([^/]+?)(?:/)?$ {
    try_files /[foo]/[bar]/[id].html /index.html;
}"
`;

exports[`generate routes for example project with trailingSlash enabled 1`] = `
"
location ~ ^/(?:/)?$ {
    try_files /index.html /index.html;
}

location ~ ^/hello(?:/)?$ {
    try_files /hello.html /index.html;
}

location ~ ^/hello/world(?:/)?$ {
    try_files /hello/world.html /index.html;
}

location ~ ^/([^/]+?)(?:/)?$ {
    try_files /[foo].html /index.html;
}

location ~ ^/([^/]+?)/([^/]+?)/hello(?:/)?$ {
    try_files /[foo]/[bar]/hello.html /index.html;
}

location ~ ^/([^/]+?)/([^/]+?)/([^/]+?)(?:/)?$ {
    try_files /[foo]/[bar]/[id].html /index.html;
}"
`;
